python/nano/example/pytorch/inference_pipeline/inference_pipeline.py

```diff
@@ -0,0 +1,319 @@
+
+    def __dataloader(self, stage="train"):
+        """Train/validation loaders."""
+        if stage == "train":
+            dataset = self.create_dataset(self.data_path.joinpath("train"), self.train_transform)
+            return DataLoader(dataset=dataset, batch_size=self._train_batch_size, num_workers=self._num_workers, shuffle=True)
+        elif stage == "val":
+            dataset = self.create_dataset(self.data_path.joinpath("validation"), self.valid_transform)
+            return DataLoader(dataset=dataset, batch_size=self._val_batch_size, num_workers=self._num_workers, shuffle=False)
+        else:
+            dataset = self.create_dataset(self.data_path.joinpath("validation"), self.valid_transform)
+            return DataLoader(dataset=dataset, batch_size=self._predict_batch_size, num_workers=self._num_workers, shuffle=False)
+
+    def train_dataloader(self):
+        log.info("Training data loaded.")
+        return self.__dataloader(stage="train")
+
+    def val_dataloader(self):
+        log.info("Validation data loaded.")
+        return self.__dataloader(stage="val")
+    
+    def predict_dataloader(self):
+        log.info("Prediction data loaded.")
+        return self.__dataloader(stage="predict")
+
+
+class MilestonesFinetuning(BaseFinetuning):
+    def __init__(self, milestones: tuple = (5, 10), train_bn: bool = False):
+        super().__init__()
+        self.milestones = milestones
+        self.train_bn = train_bn
+
+    def freeze_before_training(self, pl_module: LightningModule):
+        self.freeze(modules=pl_module.feature_extractor, train_bn=self.train_bn)
+
+    def finetune_function(self, pl_module: LightningModule, epoch: int, optimizer: Optimizer, opt_idx: int):
+        if epoch == self.milestones[0]:
+            # unfreeze 5 last layers
+            self.unfreeze_and_add_param_group(
+                modules=pl_module.feature_extractor[-5:], optimizer=optimizer, train_bn=self.train_bn
+            )
+
+        elif epoch == self.milestones[1]:
+            # unfreeze remaining layers
+            self.unfreeze_and_add_param_group(
+                modules=pl_module.feature_extractor[:-5], optimizer=optimizer, train_bn=self.train_bn
+            )
+
+
+# finetune on new dataset
+trainer = Trainer(max_epochs=15, callbacks=[MilestonesFinetuning(milestones)])
+model = TransferLearningModel(milestones=milestones)
+datamodule = CatDogImageDataModule()
+# trainer.fit(model, datamodule)
+# trainer.save(model, "best_model.ckpt")
+trainer.load(model, "best_model.ckpt")
+
+# define metric for accuracy calculation
+def accuracy(pred, target):
+    pred = torch.sigmoid(pred)
+    target = target.view((-1, 1)).type_as(pred).int()
+    return Accuracy()(pred, target)
+
+# accelaration inference using InferenceOptimizer
+model.eval()
+optimizer = InferenceOptimizer()
+optimizer.optimize(model=model,
+                   training_data=datamodule.predict_dataloader(),
```

Comment:
we need to use training dataloader here to avoid data leak as we stated here https://github.com/intel-analytics/BigDL/blob/1b45bf03830cd5c47edaf23bd78705edd5143313/python/nano/src/bigdl/nano/pytorch/inference/optimizer.py#L115-L121